cmake/extensions.cmake

```diff
@@ -852,15 +852,20 @@ endfunction()
 #    NOINIT       Inside the noinit output section.
 #    RWDATA       Inside the data output section.
 #    RODATA       Inside the rodata output section.
+#    TEXT_START   At the beginning of the text section, i.e. the beginning of
+#                 the image.
 #    RAM_SECTIONS Inside the RAMABLE_REGION GROUP.
 #    SECTIONS     Near the end of the file. Don't use this when linking into
 #                 RAMABLE_REGION, use RAM_SECTIONS instead.
+# <sort_key> is an optional key to sort by inside of each location. The key must
+#    be alphanumeric, and the keys are sorted alphabetically. If no key is
+#    given, the key 'default' is used. Keys are case-sensitive.
```

Comment:
Let's rename KEY to SORTED_BY_STRING or similar. Key has some connotations we want to avoid, e.g. it could be a key-value system or be short for keyword.

SORTED_BY_STRING, although verbose, makes it more clear that this has to do with ordering when a new reader sees the usage IMHO (see below).

```
# <sorted_by_string> is an optional string to sort by inside of each location. The string must
#    be alphanumeric, and the strings are sorted alphabetically. If no string is
#    given, the string 'default' is used. Strings are case-sensitive.
```

```
zephyr_linker_sources_ifdef(CONFIG_SW_VECTOR_RELAY
  TEXT_START
  SORTED_BY_STRING 0x0relay_vectors
  relay_vector_table.ld
  )


```

Reply:
How about `SORT_KEY`?

Reply:
fair enough